src/main/java/thrift/commons/util/DateUtils.java

```diff
@@ -0,0 +1,286 @@
+     */
+    public static boolean isAfterDay(Calendar cal1, Calendar cal2) {
+        if (cal1 == null || cal2 == null) {
+            throw new IllegalArgumentException("The dates must not be null");
+        }
+        if (cal1.get(Calendar.ERA) < cal2.get(Calendar.ERA)) {
+            return false;
+        }
+        if (cal1.get(Calendar.ERA) > cal2.get(Calendar.ERA)) {
+            return true;
+        }
+        if (cal1.get(Calendar.YEAR) < cal2.get(Calendar.YEAR)) {
+            return false;
+        }
+        if (cal1.get(Calendar.YEAR) > cal2.get(Calendar.YEAR)) {
+            return true;
+        }
+        return cal1.get(Calendar.DAY_OF_YEAR) > cal2.get(Calendar.DAY_OF_YEAR);
+    }
+
+    /**
+     * <p>Checks if a date is after today and within a number of days in the future.</p>
+     * @param date the date to check, not altered, not null.
+     * @param days the number of days.
+     * @return true if the date day is after today and within days in the future .
+     * @throws IllegalArgumentException if the date is <code>null</code>
+     */
+    public static boolean isWithinDaysFuture(Date date, int days) {
+        if (date == null) {
+            throw new IllegalArgumentException("The date must not be null");
+        }
+        Calendar cal = Calendar.getInstance();
+        cal.setTime(date);
+        return isWithinDaysFuture(cal, days);
+    }
+
+    /**
+     * <p>Checks if a calendar date is after today and within a number of days in the future.</p>
+     * @param cal the calendar, not altered, not null
+     * @param days the number of days.
+     * @return true if the calendar date day is after today and within days in the future .
+     * @throws IllegalArgumentException if the calendar is <code>null</code>
+     */
+    public static boolean isWithinDaysFuture(Calendar cal, int days) {
+        if (cal == null) {
+            throw new IllegalArgumentException("The date must not be null");
+        }
+        Calendar today = Calendar.getInstance();
+        Calendar future = Calendar.getInstance();
+        future.add(Calendar.DAY_OF_YEAR, days);
+        return (isAfterDay(cal, today) && !isAfterDay(cal, future));
+    }
+
+    /** Returns the given date with the time set to the start of the day. */
+    public static Date getStart(Date date) {
+        return clearTime(date);
+    }
+
+    /** Returns the given date with the time values cleared. */
+    public static Date clearTime(Date date) {
+        if (date == null) {
+            return null;
+        }
+        Calendar c = Calendar.getInstance();
+        c.setTime(date);
+        c.set(Calendar.HOUR_OF_DAY, 0);
+        c.set(Calendar.MINUTE, 0);
+        c.set(Calendar.SECOND, 0);
+        c.set(Calendar.MILLISECOND, 0);
+        return c.getTime();
+    }
+
+    /** Determines whether or not a date has any time values (hour, minute,
+     * seconds or millisecondsReturns the given date with the time values cleared. */
```

Comment:
Should `/*` be used instead of `/**` for normal comments as Javadoc tool will detect `/**` as Javadoc?

Reply:
Resolved in 58f47b5